lib/pbench/server/filetree.py

```diff
@@ -303,30 +316,33 @@ def unpack(self, incoming: Path, results: Path):
         Args:
             incoming: Controller's directory in the INCOMING tree
             results: Controller's directory in the RESULTS tree
+
+        Raises:
+            TarballUnpackError if an exception occurs while unpacking the tarball
         """
         unpacked = incoming / self.name
         unpacked.mkdir(parents=True)
         self.unpacked = unpacked
 
         try:
+            args = ["tar", "-xf", self.tarball_path]
             process = subprocess.run(
-                ["tar", "-xf", self.tarball_path],
+                args,
                 cwd=self.unpacked,
                 stderr=subprocess.DEVNULL,
             )
-        except Exception:
-            self.logger.exception("Failed to extract tar ball, {}", self.name)
+        except Exception as exc:
+            raise TarballUnpackError(Path(self.tarball_path), exc)
         else:
-            if process.exitcode != 0:
-                self.logger.error(
-                    "unpacking for {} failed with {} exitcode",
-                    self.name,
-                    process.exitcode,
+            if process.returncode != 0:
+                raise TarballUnpackError(
+                    Path(self.tarball_path.name),
+                    f"Subprocess returncode: {process.returncode}",
                 )
-
-        results_link = results / self.name
-        results_link.symlink_to(unpacked)
-        self.results_link = results_link
+            else:
```

Comment:
The `else` makes the flow clear, but it's also awkward to end the method nested two levels deep. And since we either get an exception and raise `TarballUnpackError` *or* detect `process.returncode != 0` and raise `TarballUnpackError`, it might be more clear to drop this `else` and un-indent the success path.